## 92% of end-end tests have passed
```
Total executed tests: 62
Failed tests: 4
Expected to fail tests: 1
Passed tests: 57
```
```
IDs of failed tests: 702782,702732,703495,703194 
```
```
IDs of expected to fail tests: 702844 
```
<h3>Failed tests (4)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=bug/Invalid-Hex-Data-for-Private-Key-and-Invalid-PIN-Code-errors-after-mobile-attempts-to-be-synced-with-Desktop-imported-on-keycard-%2321812&PR_ID=22082&APK_NAME=22082.apk&TR_CASE_IDS=702782,702732,703495,703194">Rerun failed tests</a></li><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785846">test_community_several_images_send_reply</a>, id: 703194 </b></td></tr><tr><td><p><blockquote><div>Device 2: Trying to access images (image container) inside message with text 'gallery' </div><div>Device 2: Find `ImageContainer` by `xpath`: `//*[starts-with(@text,'gallery')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']//*[@content-desc="image-container"]` </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:432: in test_community_several_images_send_reply
    if (image.is_element_differs_from_template("images_gallery.png", 5) and
../views/base_element.py:259: in is_element_differs_from_template
    difference = ImageChops.difference(self.image, self.template)
../views/base_element.py:237: in image
    return Image.open(BytesIO(base64.b64decode(self.find_element().screenshot_as_base64)))
../views/base_element.py:78: in find_element
    raise NoSuchElementException(
 Device 2: ImageContainer by xpath: `//*[starts-with(@text,'gallery')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']//*[@content-desc="image-container"]` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESgdSzTI8Pjzl1RVfz">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESAeJ9Kcx7njbzpLXO">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785812">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td><p><blockquote><div>Device 1: Looking for a pinned message by text: Message 2 </div><div>Device 1: Find `Button` by `xpath`: `//*[@content-desc='pinned-messages-menu']//*[starts-with(@text,'Message 2')]` </div></blockquote></p>

```
critical/chats/test_group_chat.py:411: in test_group_chat_pin_messages
    self.chats[0].pinned_messages_list.message_element_by_text(self.message_2).long_press_without_release()
../views/base_element.py:315: in long_press_without_release
    action.click_and_hold(self.find_element()).perform()
../views/base_element.py:78: in find_element
    raise NoSuchElementException(
 Device 1: Button by xpath: `//*[@content-desc='pinned-messages-menu']//*[starts-with(@text,'Message 2')]` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESN78uRUiEzeJvSKOz">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESd6VYr7ZSEd10ZuK2">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESDprkbI9gUw8PsPYy">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785851">test_group_chat_mute_chat</a>, id: 703495 </b></td></tr><tr><td><p><blockquote><div>Device 2: Click until `ChatMessageInput` by `accessibility id`: `chat-message-input` will be presented </div><div>Device 2: Looking for a message by text: Chat is unmuted now </div></blockquote></p>

```
critical/chats/test_group_chat.py:539: in test_group_chat_mute_chat
    self.errors.verify_no_errors()
base_test_case.py:179: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 2: Message text 'Chat is unmuted now' is not shown in chat preview after unmute
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESN78uRUiEzeJvSKOz">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESd6VYr7ZSEd10ZuK2">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESDprkbI9gUw8PsPYy">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785818">test_1_1_chat_emoji_send_reply_and_open_link</a>, id: 702782 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Button` by `accessibility id`: `reaction-thumbs-up` </div><div>Device 2: Find `Button` by `accessibility id`: `reaction-thumbs-up` </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:158: in test_1_1_chat_emoji_send_reply_and_open_link
    self.chat_2.set_reaction(url_message)
../views/chat_view.py:692: in set_reaction
    element.double_click()
../views/base_element.py:113: in double_click
    [self.find_element().click() for _ in range(2)]
../views/base_element.py:113: in <listcomp>
    [self.find_element().click() for _ in range(2)]
../views/base_element.py:78: in find_element
    raise NoSuchElementException(
 Device 2: Button by accessibility id: `reaction-thumbs-up` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESN2HHFdmvTm6wpCpW">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8jUTk6SUS1RA6wca">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Expected to fail tests (1)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785831">test_community_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 702844 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `EmojisNumber` by `xpath`: `//*[starts-with(@text,'https://m.youtube.com/watch?v=Je7yErjEVt4')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']/../..//*[@content-desc='emoji-reaction-4']/android.widget.TextView[2]` </div><div>Device 2: Element EmojisNumber text is equal to 1 </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:662: in test_community_links_with_previews_github_youtube_twitter_gif_send_enable
    self.errors.verify_no_errors()
base_test_case.py:179: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 1: No preview is loaded for url https://youtu.be/Je7yErjEVt4
E    Device 1: No preview is loaded for url https://www.youtube.com/watch?v=XN-SVmuJH2g&list=PLbrz7IuP1hrgNtYe9g6YHwHO6F3OqNMao
E    Device 1: No preview is loaded for url https://m.youtube.com/watch?v=Je7yErjEVt4 
```

<b>[[Youtube links preview is not loaded on LambdaTest emulators, needs investigation]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESgdSzTI8Pjzl1RVfz">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESAeJ9Kcx7njbzpLXO">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (57)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestActivityCenterContactRequestMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785835">test_activity_center_contact_request_accept_swipe_mark_all_as_read</a>, id: 702851 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESUXsB4mVV67OVC3ts">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES5bAXdjvOIJEBpYNK">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785834">test_activity_center_contact_request_decline</a>, id: 702850 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESUXsB4mVV67OVC3ts">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES5bAXdjvOIJEBpYNK">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785817">test_add_contact_field_validation</a>, id: 702777 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESUXsB4mVV67OVC3ts">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES5bAXdjvOIJEBpYNK">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785859">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785867">test_wallet_swap_flow_mainnet</a>, id: 741555 </b></td></tr><tr><td></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785864">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785868">test_wallet_bridge_flow_mainnet</a>, id: 741612 </b></td></tr><tr><td></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785866">test_wallet_send_flow_mainnet</a>, id: 741554 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMergedTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785832">test_community_leave</a>, id: 702845 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESe0dP2ttwzybMPPNG">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESxwg4alYSgUzzQfpf">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785821">test_community_mentions_push_notification</a>, id: 702786 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESe0dP2ttwzybMPPNG">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESxwg4alYSgUzzQfpf">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785824">test_community_markdown_support</a>, id: 702809 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESe0dP2ttwzybMPPNG">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESxwg4alYSgUzzQfpf">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785841">test_community_hashtag_links_to_community_channels</a>, id: 702948 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESe0dP2ttwzybMPPNG">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESxwg4alYSgUzzQfpf">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785854">test_community_join_when_node_owner_offline</a>, id: 703629 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESe0dP2ttwzybMPPNG">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESxwg4alYSgUzzQfpf">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785845">test_restore_multiaccount_with_waku_backup_remove_profile_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESPiszBpZOqGMqHyef">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785814">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESPiszBpZOqGMqHyef">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785833">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESPiszBpZOqGMqHyef">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785838">test_community_undo_delete_message</a>, id: 702869 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESPiszBpZOqGMqHyef">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785849">test_community_mute_community_and_channel</a>, id: 703382 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESPiszBpZOqGMqHyef">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785853">test_community_discovery</a>, id: 703503 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESPiszBpZOqGMqHyef">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePRTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785843">test_activity_center_admin_notification_accept_swipe</a>, id: 702958 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESRKOEd3bOBvlMMcgj">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESjxgsddwZ2yN3G19x">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785842">test_activity_center_mentions</a>, id: 702957 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESRKOEd3bOBvlMMcgj">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESjxgsddwZ2yN3G19x">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785871">test_activity_centre_contact_request_receiver_offline</a>, id: 741809 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESRKOEd3bOBvlMMcgj">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESjxgsddwZ2yN3G19x">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785847">test_group_chat_reactions</a>, id: 703202 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESN78uRUiEzeJvSKOz">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESd6VYr7ZSEd10ZuK2">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESDprkbI9gUw8PsPYy">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785822">test_group_chat_join_send_text_messages_push</a>, id: 702807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESN78uRUiEzeJvSKOz">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESd6VYr7ZSEd10ZuK2">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESDprkbI9gUw8PsPYy">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785823">test_group_chat_offline_pn</a>, id: 702808 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESN78uRUiEzeJvSKOz">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESd6VYr7ZSEd10ZuK2">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESDprkbI9gUw8PsPYy">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785848">test_group_chat_send_image_save_and_share</a>, id: 703297 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESN78uRUiEzeJvSKOz">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESd6VYr7ZSEd10ZuK2">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESDprkbI9gUw8PsPYy">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785840">test_activity_center_reply_read_unread_delete_filter_swipe</a>, id: 702947 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESRv9DTO30txteRrw6">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESuvUtIVcNhpuGb1e8">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785836">test_1_1_chat_edit_message</a>, id: 702855 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESN2HHFdmvTm6wpCpW">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8jUTk6SUS1RA6wca">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785810">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESN2HHFdmvTm6wpCpW">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8jUTk6SUS1RA6wca">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785815">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESN2HHFdmvTm6wpCpW">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8jUTk6SUS1RA6wca">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785811">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESN2HHFdmvTm6wpCpW">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8jUTk6SUS1RA6wca">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785813">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESN2HHFdmvTm6wpCpW">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8jUTk6SUS1RA6wca">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785825">test_1_1_chat_push_emoji</a>, id: 702813 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESN2HHFdmvTm6wpCpW">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8jUTk6SUS1RA6wca">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785850">test_1_1_chat_send_image_save_and_share</a>, id: 703391 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESN2HHFdmvTm6wpCpW">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8jUTk6SUS1RA6wca">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestFallbackMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785865">test_fallback_add_key_pair</a>, id: 741054 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785861">test_fallback_sync_with_error</a>, id: 740220 </b></td></tr><tr><td></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785862">test_fallback_with_correct_seed_phrase</a>, id: 740221 </b></td></tr><tr><td></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785863">test_fallback_validate_seed_phrase</a>, id: 740222 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestDeepLinksOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785855">test_links_open_universal_links_from_chat</a>, id: 704613 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESeDZ26EMWW2pDUsrs">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785816">test_links_deep_links_profile</a>, id: 702775 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESeDZ26EMWW2pDUsrs">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785860">test_deep_links_communities</a>, id: 739307 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESeDZ26EMWW2pDUsrs">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestAndroid12:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785869">test_create_account_android_12</a>, id: 741806 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESVYDHoUdHKzEe88dm">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785858">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785857">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestAndroid13:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785870">test_create_account_android_13</a>, id: 741807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESDbF2oZbdjobAtHuy">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUiTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785852">test_1_1_chat_mute_chat</a>, id: 703496 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8J2YnS6NKWRXFPBD">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES3FTh99SKyWBDdkhj">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785819">test_1_1_chat_is_shown_message_sent_delivered_from_offline</a>, id: 702783 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8J2YnS6NKWRXFPBD">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES3FTh99SKyWBDdkhj">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785820">test_1_1_chat_delete_via_long_press_relogin</a>, id: 702784 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8J2YnS6NKWRXFPBD">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES3FTh99SKyWBDdkhj">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785828">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESgdSzTI8Pjzl1RVfz">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESAeJ9Kcx7njbzpLXO">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785839">test_community_contact_block_unblock_offline</a>, id: 702894 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESgdSzTI8Pjzl1RVfz">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESAeJ9Kcx7njbzpLXO">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785844">test_community_mark_all_messages_as_read</a>, id: 703086 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESgdSzTI8Pjzl1RVfz">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESAeJ9Kcx7njbzpLXO">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785829">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESgdSzTI8Pjzl1RVfz">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESAeJ9Kcx7njbzpLXO">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785827">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESgdSzTI8Pjzl1RVfz">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESAeJ9Kcx7njbzpLXO">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785826">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESgdSzTI8Pjzl1RVfz">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESAeJ9Kcx7njbzpLXO">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785856">test_community_edit_delete_message_when_offline</a>, id: 704615 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESgdSzTI8Pjzl1RVfz">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESAeJ9Kcx7njbzpLXO">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785837">test_community_one_image_send_reply</a>, id: 702859 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESgdSzTI8Pjzl1RVfz">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESAeJ9Kcx7njbzpLXO">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3785830">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESgdSzTI8Pjzl1RVfz">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESAeJ9Kcx7njbzpLXO">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/status_im/common/pairing/events.cljs
+++ b/src/status_im/common/pairing/events.cljs
@@ -9,35 +9,38 @@
   [{:keys [db]} {:keys [type action data error] :as event}]
   (log/info "local pairing signal received"
             {:event event})
-  (let [{:keys [account password]}      data
-        role                            (get-in db [:syncing :role])
-        receiver?                       (= role constants/local-pairing-role-receiver)
-        sender?                         (= role constants/local-pairing-role-sender)
-        connection-success?             (and (= type
-                                                constants/local-pairing-event-connection-success)
-                                             (= action
-                                                constants/local-pairing-action-connect))
-        connection-error?               (= type
-                                           constants/local-pairing-event-connection-error)
-        error-on-pairing?               (contains? constants/local-pairing-event-errors type)
-        completed-pairing?              (and (= type
-                                                constants/local-pairing-event-transfer-success)
-                                             (= action
-                                                constants/local-pairing-action-pairing-installation))
-        received-account?               (and (= type
-                                                constants/local-pairing-event-received-account)
-                                             (= action
-                                                constants/local-pairing-action-pairing-account)
-                                             (and (some? account) (some? password)))
-        multiaccount-data               (when received-account?
-                                          (merge account {:password password}))
-        navigate-to-syncing-devices?    (and (or connection-success? connection-error?) receiver?)
-        user-in-syncing-devices-screen? (or (= (:view-id db) :screen/onboarding.syncing-progress)
-                                            (= (:view-id db) :screen/profile.profiles)
-                                            (= (:view-id db) :screen/onboarding.syncing-progress-intro))
-        user-in-sign-in-intro-screen?   (= (:view-id db) :screen/onboarding.sign-in-intro)
-        keystore-files-transfer-action? (= action
-                                           constants/local-pairing-action-keystore-files-transfer)]
+  (let [{:keys [account password chatKey]} data
+        role                               (get-in db [:syncing :role])
+        receiver?                          (= role constants/local-pairing-role-receiver)
+        sender?                            (= role constants/local-pairing-role-sender)
+        connection-success?                (and (= type
+                                                   constants/local-pairing-event-connection-success)
+                                                (= action
+                                                   constants/local-pairing-action-connect))
+        connection-error?                  (= type
+                                              constants/local-pairing-event-connection-error)
+        error-on-pairing?                  (contains? constants/local-pairing-event-errors type)
+        completed-pairing?                 (and (= type
+                                                   constants/local-pairing-event-transfer-success)
+                                                (= action
+                                                   constants/local-pairing-action-pairing-installation))
+        received-account?                  (and (= type
+                                                   constants/local-pairing-event-received-account)
+                                                (= action
+                                                   constants/local-pairing-action-pairing-account)
+                                                (and (some? account) (some? password)))
+        multiaccount-data                  (when received-account?
+                                             (merge account
+                                                    {:password            password
+                                                     :whisper-private-key chatKey}))
+        navigate-to-syncing-devices?       (and (or connection-success? connection-error?) receiver?)
+        user-in-syncing-devices-screen?    (or (= (:view-id db) :screen/onboarding.syncing-progress)
+                                               (= (:view-id db) :screen/profile.profiles)
+                                               (= (:view-id db)
+                                                  :screen/onboarding.syncing-progress-intro))
+        user-in-sign-in-intro-screen?      (= (:view-id db) :screen/onboarding.sign-in-intro)
+        keystore-files-transfer-action?    (= action
+                                              constants/local-pairing-action-keystore-files-transfer)]
     (merge {:db (cond-> db
                   connection-success?
                   (assoc-in [:syncing :pairing-status] :connected)

--- a/src/status_im/contexts/profile/login/events.cljs
+++ b/src/status_im/contexts/profile/login/events.cljs
@@ -26,16 +26,22 @@
 
 (rf/reg-event-fx :profile.login/local-paired-user
  (fn [{:keys [db]}]
-   (let [{:keys [key-uid password]} (get-in db [:syncing :profile])
-         login-sha3-password        (get-in db [:syncing :login-sha3-password])
-         password                   (if-not (nil? login-sha3-password) ;; already logged in
-                                      login-sha3-password
-                                      password)
-         masked-password            (security/mask-data password)]
-     {:db                    (-> db
-                                 (assoc-in [:onboarding/profile :password] masked-password)
-                                 (assoc-in [:onboarding/profile :syncing?] true))
-      :effects.profile/login [key-uid password]})))
+   (let [{:keys [key-uid password keycard-pairing
+                 whisper-private-key]} (get-in db [:syncing :profile])
+         login-sha3-password           (get-in db [:syncing :login-sha3-password])
+         password                      (if-not (nil? login-sha3-password) ;; already logged in
+                                         login-sha3-password
+                                         password)
+         masked-password               (security/mask-data password)]
+     {:db (-> db
+              (assoc-in [:onboarding/profile :password] masked-password)
+              (assoc-in [:onboarding/profile :syncing?] true))
+      :fx [(if keycard-pairing
+             [:effects.keycard/login-with-keycard
+              {:password            password
+               :whisper-private-key whisper-private-key
+               :key-uid             key-uid}]
+             [:effects.profile/login [key-uid password]])]})))
 
 ;; login phase 1: we want to load and show chats faster, so we split login into 2 phases
 (rf/reg-event-fx :profile.login/login-existing-profile

--- a/status-go-version.json
+++ b/status-go-version.json
@@ -3,7 +3,7 @@
     "_comment": "Instead use: scripts/update-status-go.sh <rev>",
     "owner": "status-im",
     "repo": "status-go",
-    "version": "v10.3.0",
-    "commit-sha1": "6132a9e73bba64949632f49bbf3e315054c0397b",
-    "src-sha256": "1r81qhrvbcq6q1fp3lk5knrp668cjj7k4a1pxzgzffj0lwkxys1b"
+    "version": "v10.4.0",
+    "commit-sha1": "1bfb0cef022afc1484a410cb3e0097e7523bb705",
+    "src-sha256": "13cg07ld319g1fyaqpkjnpdw6iap389dwzwnh9g870ci8scpdy5m"
 }

